src/server.ts

```diff
@@ -0,0 +1,528 @@
+import { dLog } from './logging'
+import axios from 'axios'
+import { GetServerStatusResponse, HeartbeatIndex, ScoreReport } from './types'
+
+const PlayerIDProp = "pid"
+const SlipIDProp = "sid"
+
+export default class ArcadeServerSDK {
+
+  // TODO: Update to correct port
+  url = "http://localhost:8080"
+  debug = process.env['ARCADE_DEBUG'] === '1' // debug logging
+  playerToken = "" // get from connecting player
+  heartbeatIndex: HeartbeatIndex = {}
+  poolHeartbeat?: NodeJS.Timer
+
+  constructor() {
+    dLog("starting arcade server sdk")
+  }
+
+  async getServerStatus(): Promise<GetServerStatusResponse> {
+    try {
+      let statuscode = 0
+      let retryCount = 0
+      let status: GetServerStatusResponse | null = null
+      while (statuscode !== 200) {
+        if (retryCount > 0) {
+          await new Promise((r) => {
+            setTimeout(() => {
+              r(null)
+            }, 200 * retryCount)
+          })
+        }
+        const res = await axios.post(this.url+'/api/server')
+        status = res.data
+        statuscode = res.status
+        if (statuscode > 299 && statuscode < 500) {
+          throw new Error(`server returned ${statuscode} - ${res.statusText}`)
+        }
+
+        retryCount++
+        if (retryCount >= 5) {
+          throw new Error(`max retries hit, exiting with status code ${statuscode}`)
+        }
+      }
+      return status as GetServerStatusResponse
+    } catch (error) {
+      throw error
+    }
+  }
+
+  /**
+   * Shuts down the game server. The hypervisor will send a termination signal to the server shortly after.
+   */
+  async shutdown()  {
+    try {
+      let statuscode = 0
+      let retryCount = 0
+      let status: GetServerStatusResponse | null = null
+      while (statuscode !== 200) {
```

Comment:
for all calls. Some calls return e.g. 201
```suggestion
      while (statuscode < 200 || statuscode >= 300) {
```